prefect_github/repository.py

```diff
@@ -23,6 +31,121 @@
+        description="An optional GitHubCredentials block for using private GitHub repos.",  # noqa: E501
+    )
+
+    @validator("credentials")
+    def _ensure_credentials_go_with_https(cls, v: str, values: dict):
+        """Ensure that credentials are not provided with 'SSH' formatted GitHub URLs."""
+        if v is not None:
+            if not values["repository"].startswith("https://"):
+                raise InvalidRepositoryURLError(
+                    (
+                        "Crendentials can only be used with GitHub repositories "
+                        "using the 'HTTPS' format. You must either remove the "
+                        "credential if you wish to use the 'SSH' format and are not "
+                        "using a private repository, or you must change the repository "
+                        "url to the 'HTTPS' format. "
+                    )
+                )
+
+        return v
+
+    def _create_repo_url(self) -> str:
+        """Format the URL provided to the `git clone` command.
+
+        For private repos: https://<oauth-key>@github.com/<username>/<repo>.git
+        All other repos should be the same as `self.repository`.
+        """
+
+        if self.repository.startswith("https://") and self.credentials is not None:
+            repo_url = self.repository[8:]
+            token_value = self.credentials.token.get_secret_value()
+            full_url = f"https://{token_value}@{repo_url}"
+        else:
+            full_url = self.repository
+
+        return full_url
+
+    @staticmethod
+    def _get_paths(
+        dst_dir: Union[str, None], src_dir: str, sub_directory: str
+    ) -> Tuple[str, str]:
+        """Returns the fully formed paths for GitHubRepository contents in the form
+        (content_source, content_destination).
+        """
+        if dst_dir is None:
+            content_destination = Path(".").absolute()
+        else:
+            content_destination = Path(dst_dir)
+
+        content_source = Path(src_dir)
+
+        if sub_directory:
+            content_destination = content_destination.joinpath(sub_directory)
+            content_source = content_source.joinpath(sub_directory)
+
+        return str(content_source), str(content_destination)
+
+    @sync_compatible
+    async def get_directory(
+        self, from_path: Optional[str] = None, local_path: Optional[str] = None
+    ) -> None:
+        """
+        Clones a GitHub project specified in `from_path` to the provided `local_path`;
+        defaults to cloning the repository reference configured on the Block to the
+        present working directory.
+
+        Args:
+            from_path: If provided, interpreted as a subdirectory of the underlying
+                repository that will be copied to the provided local path.
+            local_path: A local path to clone to; defaults to present working directory.
+        """
+        # CONSTRUCT COMMAND
+        cmd = f"git clone {self._create_repo_url()}"
```

Comment:
Do we want to clone with a depth of 1 here too? It'd decrease pull time and I don't think we need the git history for anything.

Reply:
That seems reasonable to me